### PR TITLE
Adds empty dashboard page

### PR DESCRIPTION
**Pull request by kaituo**
*Issue #, if available:*

*Description of changes:*

Screen Shot 2020-03-22 at 9 13 19 PM

The dashboard page is the first page that customers see when they click Anomaly detection in the side navigation.  If there is at least one detector, we show the detailed dashboard page. Otherwise, we show an empty dashboard page. This CR adds the empty dashboard page. This CR also adds logic for redirecting to a detailed or empty dashboard page.

Testing done:
- manual testing to verify the workflow
- Add snapshot unit test for the empty dashboard page.
- Add unit test for redirect logic

Note: The newly added unit tests and some other 10 test suites fail after the recent commit adds @babel/transform-runtime. Without that commit, all tests pass. Will work with the commit author to address that before pushing the PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+          }}
+        >
+          <EuiSideNav style={{ width: 150 }} items={sideNav} />
+        </EuiPageSideBar>
+        <EuiPageBody>
+          <EuiPageHeader>
+            <EuiTitle size="l">
+              <h1>Dashboard</h1>
+            </EuiTitle>
+          </EuiPageHeader>
```
Comment (yizheliu-amazon) [inline public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: I think you can move `sidebar` and `Dashboard` header to `Dashboard.tsx`, because those 2 components will be reused even in Dashboard with detectors.

Comment (kaituo) [reply public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: good idea.  moved to a util class since I don't want EmptyDashboard to reference components defined in Dashboard.tsx as there would be circular dependency.  Dashboard.tsx references EmptyDashboard now.

Comment (yizheliu-amazon) [reply public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: hmmm, my previous thought is: `Dashboard.tsx` has `sidebar` + `header` + `EmptyDashboardMessage`, and `EmptyDashboardMessage` just returns <EuiPageBody> with all the messages inside.  If there exists detectors: `Dashboard.tsx` will be `sidebar` + `header` + `ActiveDetectorsDashboard`.

Given the latest commit, since we just need to have `sidebar` + `header` in the future for active dashboard, it is fine with me. I will approve it.

File: public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (yizheliu-amazon) [inline public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: 1 more question: should we also update the copyright year to 2020 in README.md? Is there any guideline on this?

File: public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            title={<h2>You have no detectors</h2>}
+            body={
+              <Fragment>
+                <p>Create detector first to detect anomalies in your data.</p>
+                <p>
+                  Dashboard will generate insights on the anomalies across all
+                  of your detectors
+                </p>
+                <p>
+                  Read about{' '}
```
Comment (mihirsoni) [inline public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: Remove this empty spaces.

File: public/pages/Dashboard/Components/utils/common.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  EuiPageSideBar,
+  EuiPageHeader,
+  EuiTitle,
+} from '@elastic/eui';
+
+type SideBarState = {
+  selectedItemName: string;
+};
+
+export class SideBar extends Component<{}, SideBarState> {
```
Comment (mihirsoni) [inline public/pages/Dashboard/Components/utils/common.tsx]: Match the file name and class name.

File: public/pages/Dashboard/Container/Dashboard.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+  const totalDetectors = useSelector(
+    (state: AppState) => state.ad.totalDetectors
+  );
+
+  useEffect(() => {
+    onRefreshPage();
+  });
+
+  return totalDetectors == 0 ? (
```
Comment (mihirsoni) [inline public/pages/Dashboard/Container/Dashboard.tsx]: You may also have loading state, else this will trigger flash for user.
Show lower in case of `totalDetector ===0 && isLoading == true`

File: public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import { APP_PATH, PLUGIN_NAME } from '../../../../utils/constants';
+import { SideBar, DashboardHeader } from '../utils/common';
+
+export class EmptyDashboard extends Component<{}, EmptyDashboardState> {
+  render() {
+    return (
+      <EuiPage style={{ flex: 1 }}>
+        <SideBar />
+        <EuiPageBody>
+          <DashboardHeader />
```
Comment (mihirsoni) [inline public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: This component should be just EmptyDashboard, I would suggest you to move rest of the things to actual Page.

File: public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx (hunk 2 above)
Comment (kaituo) [reply public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: not sure.  I updated anyway.

File: public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx (hunk 3 above)
Comment (kaituo) [reply public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: We need the empty space.  Otherwise it would be like

Screen Shot 2020-03-24 at 7 31 40 PM

File: public/pages/Dashboard/Components/utils/common.tsx (hunk 4 above)
Comment (kaituo) [reply public/pages/Dashboard/Components/utils/common.tsx]: moved these class to different files.

File: public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx (hunk 6 above)
Comment (kaituo) [reply public/pages/Dashboard/Components/EmptyDashboard/EmptyDashboard.tsx]: Moved.

File: public/pages/Dashboard/Container/Dashboard.tsx (hunk 5 above)
Comment (kaituo) [reply public/pages/Dashboard/Container/Dashboard.tsx]: good idea.  changed to show Spanner while loading.

File: public/pages/Dashboard/Components/utils/SideBar.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+type SideBarState = {
+  selectedItemName: string;
+};
+
+export class SideBar extends Component<{}, SideBarState> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      selectedItemName: 'Lion stuff',
```
Comment (mihirsoni) [inline public/pages/Dashboard/Components/utils/SideBar.tsx]: Typo ?

File: public/pages/Dashboard/Container/Dashboard.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        <DashboardHeader />
+        {isLoading == true ? (
+          <div>
+            <EuiLoadingSpinner size="s" />
+            &nbsp;&nbsp;
+            <EuiLoadingSpinner size="m" />
+            &nbsp;&nbsp;
+            <EuiLoadingSpinner size="l" />
+            &nbsp;&nbsp;
+            <EuiLoadingSpinner size="xl" />
```
Comment (mihirsoni) [inline public/pages/Dashboard/Container/Dashboard.tsx]: Why do we need 4 spinners and not just 1 ?

File: public/pages/Dashboard/Components/utils/SideBar.tsx (hunk 7 above)
Comment (kaituo) [reply public/pages/Dashboard/Components/utils/SideBar.tsx]: Not a typo.  Just initialize the selected item to some name.  When a user selects a link, this field would be assigned to another value.

File: public/pages/Dashboard/Container/Dashboard.tsx (hunk 8 above)
Comment (kaituo) [reply public/pages/Dashboard/Container/Dashboard.tsx]: I am following the spinner example here: https://elastic.github.io/eui/#/display/loading

Four spinners with different size together look more like loading than a single spinner.
